docs/requirements/ADP-957.md

```diff
@@ -0,0 +1,156 @@
+# Basic definitions
+
+*   **Collateral inputs**, in a transaction, are used to cover transaction fees
+    in the case that a non-native script fails. The term **collateral** refers
+    to the total ada contained in the UTxOs referenced by collateral inputs.
+    Collateral inputs can **only** contain ada: they may **not** contain
+    non-ada assets.
+
+*   The protocol parameter `collateralPercent` is used to specify the
+    percentage of the total transaction fee its collateral must (at a minimum)
+    cover. The collateral inputs must not themselves be locked by a script.
+
+*   The protocol parameter `maxCollateralInputs` is used to limit,
+    additionally, the total number of collateral inputs, and thus the total
+    number of additional signatures that must be checked during validation.
+
+# Proposed requirements
+
+1.  The wallet, when requested to create a transaction requiring collateral,
+    should reserve one or more pure-ada UTxO entries to serve as collateral
```

Comment:
why more than one? I mean is it possible to have more than 1 collateral utxo required for transaction or 1 should be enough?

Reply:
@piotr-iohk wrote:
> why more than one? I mean is it possible to have more than 1 collateral utxo required for transaction or 1 should be enough?

A transaction body allows `[0 .. n]` collateral inputs. See https://input-output-hk.github.io/cardano-node/cardano-api/Cardano-Api-TxBody.html#t:TxInsCollateral

In practice, the upper bound is limited by the `maxCollInputs` protocol parameter. See
https://github.com/input-output-hk/cardano-ledger-specs/blob/bdc953e0b5615c2f86593630235174fd7c0f8c38/alonzo/formal-spec/protocol-parameters.tex#L105

Reply:
In some cases, one collateral input might be enough.

But for transactions with higher fees, we might need to select more than one collateral input.